Sources/MockoloFramework/Utils/SwiftSyntaxExtensions.swift

```diff
@@ -0,0 +1,352 @@
+                            aliasMap?[k] = v
+                        }
+                    }
+                }
+            }
+            return AnnotationMetadata(typealiases: aliasMap)
+        }
+        return nil
+    }
+    func annotationMetadata(with annotation: String) -> AnnotationMetadata? {
+        var ret: AnnotationMetadata?
+        for i in 0..<count {
+            let trivia = self[i]
+            switch trivia {
+            case .docLineComment(let val):
+                ret = metadata(with: annotation, in: val)
+                if ret != nil {
+                    return ret
+                }
+            case .docBlockComment(let val):
+                ret = metadata(with: annotation, in: val)
+                if ret != nil {
+                    return ret
+                }
+            default:
+                continue
+            }
+        }
+        return nil
+    }
+}
+
+extension Syntax {
+    var name: String {
+        if let decl = self as? ProtocolDeclSyntax {
+            return decl.identifier.description.trimmingCharacters(in: .whitespaces)
+        }
+        if let decl = self as? ClassDeclSyntax {
+            return decl.identifier.description.trimmingCharacters(in: .whitespaces)
+        }
+        return .unknownVal
+    }
+    
+    var isProtocol: Bool {
+        return self is ProtocolDeclSyntax
+    }
+    
+    var isClass: Bool {
+        return self is ClassDeclSyntax
+    }
+    
+    func annotationMetadata(with annotation: String) -> AnnotationMetadata? {
+        return leadingTrivia?.annotationMetadata(with: annotation)
+    }
+    
+    // MARK - update the following
+    var offset: Int64 {
+        return Int64(self.position.utf8Offset)
+    }
+    
+    var length: Int64 {
+        return Int64(self.totalLength.utf8Length)
+    }
+}
+
+
+class EntityVisitor: SyntaxVisitor {
+    var topLevel: Bool = false
+    var entities: [Entity] = []
+    var current: Entity?
+    var imports: [String] = []
+    let annotation: String
+    
+    init(annotation: String = "") {
+        self.annotation = annotation
+    }
+    
+    func reset() {
+        topLevel = false
+        current = nil
+        entities = []
+        imports = []
+    }
+    
+    func visit(_ node: MemberDeclBlockSyntax) -> SyntaxVisitorContinueKind {
+        topLevel = false
+        return .visitChildren
+    }
+    
+    func visit(_ node: DeclModifierSyntax) -> SyntaxVisitorContinueKind {
+        if topLevel {
+            current?.acl = node.name.text
+        }
+        return .visitChildren
+    }
+    
+    func visit(_ node: AttributeListSyntax) -> SyntaxVisitorContinueKind {
+        current?.attributes.append(node.withoutTrivia().description.trimmingCharacters(in: .whitespacesAndNewlines))
+        return .visitChildren
+    }
+    
+    func visit(_ node: VariableDeclSyntax) -> SyntaxVisitorContinueKind {
+        for v in node.bindings {
```

Comment:
some comments on what is being done here may be helpful.

Reply:
ditto ^